File: android/app/src/main/java/com/github/makewheels/video2022/data/model/WatchInfo.kt
package com.github.makewheels.video2022.data.model

data class WatchInfo(
    val videoId: String,
    val coverUrl: String?,
    val videoStatus: String?,
    val multivariantPlaylistUrl: String?,
    val progressInMillis: Long?
)

data class HeartbeatRequest(
    val videoId: String,
    val clientId: String,
    val sessionId: String,
    val videoStatus: String = "READY",
    val playerProvider: String = "ANDROID_EXOPLAYER",
    val clientTime: String,
    val type: String = "TIMER",
    val event: String? = null,
    val playerTime: Long,
    val playerStatus: String,
    val playerVolume: Float
)

data class ProgressResponse(
    val progressInMillis: Long?
)

File: ios/VideoApp/Sources/Models/WatchInfo.swift
import Foundation

struct WatchInfo: Decodable {
    let videoId: String
    let coverUrl: String?
    let videoStatus: String?
    let multivariantPlaylistUrl: String?
    let progressInMillis: Int64?
}

struct HeartbeatRequest: Encodable {
    let videoId: String
    let clientId: String
    let sessionId: String
    let videoStatus: String
    let playerProvider: String
    let clientTime: String
    let type: String
    let event: String?
    let playerTime: Int64
    let playerStatus: String
    let playerVolume: Float
}

struct ProgressResponse: Decodable {
    let progressInMillis: Int64?
}
